### PR TITLE
chore!: update the license to AGPLv3 or Commercial

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This updates the license of this repo and adds an ADR to supersede the last one.

:warning: **BREAKING CHANGE** - once this is merged any repos using the lint workflow will break until they run `uds run lint:fix-license` to add the `AGPL-3.0-or-later OR Commercial` SPDX Identifier - this may also require some adjustment to things like Helm templates

## Checklist before merging

- [X] ADR proposed if making an architectural change to the repo
- [X] Tests run, docs added or updated as needed

### CODE REVIEW

File: docs/adrs/0004-agpl-v3-for-all-uds-products.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# 4. [Affero General Public License v3 (AGPLv3)](https://opensource.org/license/agpl-v3) for All UDS Products
```
Comment (mjnagel) [inline docs/adrs/0004-agpl-v3-for-all-uds-products.md:1]: Do we need the title to reflect the dual license? I don't have a strong leaning either way, just wanted to ask.

**Review by mjnagel — APPROVED** [review]
❤️

**Comment by Racer159** [comment]
First commit the linter fails successfully: https://github.com/defenseunicorns/uds-common/actions/runs/11261472517/job/31314991974?pr=286

File: docs/adrs/0004-agpl-v3-for-all-uds-products.md (hunk 1 above)
Comment (Racer159) [reply docs/adrs/0004-agpl-v3-for-all-uds-products.md]: added that for clarity

**Review by zachariahmiller — DISMISSED** [review]

**Comment by zachariahmiller** [comment]
>  this may also require some adjustment to things like Helm templates
😢

**Comment by eddiezane** [comment]
I don't think `Commercial` is a valid spdx identifier. I think we need to do something like this

https://www.qt.io/blog/switching-to-spdx#:~:text=//%20SPDX%2DLicense%2DIdentifier%3A%20LicenseRef%2DQt%2DCommercial%20OR%20LGPL%2D3.0%2Donly%20OR%20GPL%2D2.0%2Donly%20OR%20GPL%2D3.0%2Donly

**Review by mjnagel — APPROVED** [review]

**Review by andrewg-xyz — DISMISSED** [review]

**Review by daveworth — DISMISSED** [review]

**Review by andrewg-xyz — DISMISSED** [review]

**Review by jeff-mccoy — APPROVED** [review]

**Review by mjnagel — APPROVED** [review]

**Pull request merged**